### PR TITLE
Enabling a variable prefix for renamed files

### PR DESCRIPTION
**Pull request by manuerwin**
Thanks for this great rename script!
I’ve updated it slightly to allow a variable file name prefix instead of GoPro.

### CODE REVIEW

File: gopro-rename [hunk 1]
```diff
@@ -27,54 +27,55 @@ import ntpath
...
 import sys
 
-def GetRenamedFilename(path):
+def GetRenamedFilename(path, prefix):
     dir, file = ntpath.split(path)
 
     match = re.match('GOPR(\d{4})(\..*)', file)
     if match:
-        return os.path.join(dir, 'GoPro_%04d_%02d%s' % (int(match.group(1)), 0, match.group(2)))
+        return os.path.join(dir, prefix + '_%04d_%02d%s' % (int(match.group(1)), 0, match.group(2)))
```
Comment (AMDmi3) [inline gopro-rename]: `'%s_...' % (prefix, ...)`, so it's safe and consistent

Comment (AMDmi3) [reply gopro-rename]: Also maybe include `_` into prefix so it's fully customizable.

File: gopro-rename [hunk 2]
```diff
@@ -27,21 +27,22 @@ import ntpath
...
 
     match = re.match('GP(\d{2})(\d{4})(\..*)', file)
     if match:
-        return os.path.join(dir, 'GoPro_%04d_%02d%s' % (int(match.group(2)), int(match.group(1)), match.group(3)))
+        return os.path.join(dir, prefix + '_%04d_%02d%s' % (int(match.group(2)), int(match.group(1)), match.group(3)))
```
Comment (AMDmi3) [inline gopro-rename]: Likewise

File: gopro-rename [hunk 3]
```diff
@@ -50,31 +51,31 @@ def TryRename(path, dryrun, force, verbose):
...
     parser.add_argument('-r', '-R', '--recursive', action='store_true', help='process subdirectories recursively')
+    parser.add_argument('-p', '--prefix', nargs='?', default='GoPro', help='optional prefix instead of the default "GoPro"')
     parser.add_argument('paths', metavar='path', nargs='+', help='file or directory to process')
 
     args = parser.parse_args()
 
     for path in args.paths:
-        Process(path, args.dry_run, args.force, args.verbose, args.recursive)
-
+        Process(path, args.dry_run, args.force, args.verbose, args.prefix, args.recursive)
```
Comment (AMDmi3) [inline gopro-rename]: IMO it'd be better to change this to passing `args` around instead of a bunch of args.

File: gopro-rename [hunk 4]
```diff
@@ -50,31 +51,31 @@ def TryRename(path, dryrun, force, verbose):
...
-                Process(os.path.join(path, entry), dryrun, force, verbose, recursive)
+                Process(os.path.join(path, entry), dryrun, force, verbose, prefix, recursive)
 
 def Main():
     parser = argparse.ArgumentParser(description='Goes through files or directories specified on command line and renames files produced by GoPro cameras in a more usable and sortable way')
     parser.add_argument('-n', '--dry-run', action='store_true', help='do not actually rename anything')
     parser.add_argument('-f', '--force', action='store_true', help='rename even if target file exists')
     parser.add_argument('-v', '--verbose', action='store_true', help='print actions to stderr')
     parser.add_argument('-r', '-R', '--recursive', action='store_true', help='process subdirectories recursively')
+    parser.add_argument('-p', '--prefix', nargs='?', default='GoPro', help='optional prefix instead of the default "GoPro"')
```
Comment (AMDmi3) [inline gopro-rename]: I'd change help string to not duplicate default prefix value. Just `name prefix for renamed files`, or something more nicely worded.

**Review by AMDmi3 — CHANGES_REQUESTED** [review]

File: gopro-rename (hunk 1 above)
Comment (manuerwin) [reply gopro-rename]: Thanks, will do.

File: gopro-rename (hunk 3 above)
Comment (manuerwin) [reply gopro-rename]: I'll take a look and see what I can do. Still early days with Python :)

Comment (manuerwin) [reply gopro-rename]: Done

File: gopro-rename (hunk 4 above)
Comment (manuerwin) [reply gopro-rename]: Done

File: gopro-rename (hunk 2 above)
Comment (manuerwin) [reply gopro-rename]: Done

File: gopro-rename (hunk 1 above)
Comment (manuerwin) [reply gopro-rename]: Done

File: README.md [hunk 5]
```diff
@@ -30,7 +30,7 @@ GoPro_0003_01.MP4
 ## Usage
 
 ```
-gopro-rename [-nfvr] path [path ...]
```
Comment (AMDmi3) [inline README.md:33]: Would be nice to have it as `gopro-rename [-nfvr] [-p prefix] path [path...]`

File: gopro-rename [hunk 6]
```diff
@@ -27,54 +27,57 @@ import ntpath
...
     parser = argparse.ArgumentParser(description='Goes through files or directories specified on command line and renames files produced by GoPro cameras in a more usable and sortable way')
     parser.add_argument('-n', '--dry-run', action='store_true', help='do not actually rename anything')
     parser.add_argument('-f', '--force', action='store_true', help='rename even if target file exists')
     parser.add_argument('-v', '--verbose', action='store_true', help='print actions to stderr')
     parser.add_argument('-r', '-R', '--recursive', action='store_true', help='process subdirectories recursively')
+    parser.add_argument('-p', '--prefix', nargs='?', default='GoPro_', help='prefix for renamed files')
     parser.add_argument('paths', metavar='path', nargs='+', help='file or directory to process')
 
     args = parser.parse_args()
+    args_dict = vars(args)
```
Comment (AMDmi3) [inline gopro-rename]: You can just pass `args` around

**Review by AMDmi3 — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (manuerwin) [reply README.md:33]: Sure that makes sense, I've made this update.

File: gopro-rename (hunk 6 above)
Comment (manuerwin) [reply gopro-rename]: Ahhh right, rookie mistake, thanks for the advice.
Have changed it back.

**Review by AMDmi3 — APPROVED** [review]

**Comment by AMDmi3** [comment]
Thank you! :+1:

**Pull request merged**